### PR TITLE
Fix Go 1.10 build errors

### LINKED ISSUE #857: cannot use nil as type _Ctype_CFDataRef in assignment - cfssl 1.3.0 go1.10 macos
originally installed version
```
❯ cfssl version
Version: 1.2.0
Revision: dev
Runtime: go1.9.2
```

current go version
```
❯ go version
go version go1.10 darwin/amd64
```

while trying to install install new version, getting these:
```
❯ go get -u github.com/cloudflare/cfssl/cmd/cfssl
# github.com/cloudflare/cfssl/vendor/github.com/google/certificate-transparency-go/x509
projects/gohome/src/github.com/cloudflare/cfssl/vendor/github.com/google/certificate-transparency-go/x509/root_cgo_darwin.go:210: cannot use nil as type _Ctype_CFDataRef in assignment
projects/gohome/src/github.com/cloudflare/cfssl/vendor/github.com/google/certificate-transparency-go/x509/root_cgo_darwin.go:211: cannot use nil as type _Ctype_CFDataRef in assignment
projects/gohome/src/github.com/cloudflare/cfssl/vendor/github.com/google/certificate-transparency-go/x509/root_cgo_darwin.go:221: cannot convert nil to type _Ctype_CFDataRef
❯ go get -u github.com/cloudflare/cfssl/cmd/cfssljson
# github.com/cloudflare/cfssl/vendor/github.com/google/certificate-transparency-go/x509
projects/gohome/src/github.com/cloudflare/cfssl/vendor/github.com/google/certificate-transparency-go/x509/root_cgo_darwin.go:210: cannot use nil as type _Ctype_CFDataRef in assignment
projects/gohome/src/github.com/cloudflare/cfssl/vendor/github.com/google/certificate-transparency-go/x509/root_cgo_darwin.go:211: cannot use nil as type _Ctype_CFDataRef in assignment
projects/gohome/src/github.com/cloudflare/cfssl/vendor/github.com/google/certificate-transparency-go/x509/root_cgo_darwin.go:221: cannot convert nil to type _Ctype_CFDataRef
```

Works fine on Fedora 27 x64 go1.10

### PR DESCRIPTION
**Pull request by cbroglie**
Fixes #857 and #861.

Also includes switching to `dep` for dependency management.

### CODE REVIEW

**Comment by lziest** [comment]
LGTM

File: bundler/bundler.go [hunk 1]
```diff
@@ -103,14 +127,19 @@ func NewBundler(caBundleFile, intBundleFile string) (*Bundler, error) {
...
-	return NewBundlerFromPEM(caBundle, intBundle)
+	return NewBundlerFromPEM(caBundle, intBundle, opt...)
 
 }
 
 // NewBundlerFromPEM creates a new Bundler from PEM-encoded root certificates and
 // intermediate certificates.
 // If caBundlePEM is nil, the resulting Bundler can only do "Force" bundle.
-func NewBundlerFromPEM(caBundlePEM, intBundlePEM []byte) (*Bundler, error) {
+func NewBundlerFromPEM(caBundlePEM, intBundlePEM []byte, opt ...Option) (*Bundler, error) {
```
Comment (lziest) [inline bundler/bundler.go]: not sure about the semantic of a list of opt functions

Comment (cbroglie) [reply bundler/bundler.go]: This is emerging as a common pattern for initializers in Go. Some background: https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis.

And my implementation was inspired (read: mostly copied) from gRPC: https://github.com/grpc/grpc-go/blob/583a6303969ea5075e9bd1dc4b75805dfe66989a/server.go#L328.

**Comment by cbroglie** [comment]
The bundler related build failures are surprising to me, as the tests pass locally and some of them passed on Travis. I'm going to close and re-open this PR to trigger another Travis build.

File: bundler/bundler.go (hunk 1 above)
Comment (lziest) [reply bundler/bundler.go]: cool, I learn something new!

**Comment by cbroglie** [comment]
Oh, duh. Of course this fixes the Go 1.10 builds but breaks the other versions.

**Comment by cbroglie** [comment]
Ok, all the tests pass now (with the exception of the flapping macOS one).

**Pull request merged**